go-controller/pkg/node/vrfmanager/vrf_manager.go

```diff
@@ -202,13 +206,17 @@ func (vrfm *Controller) sync(vrf *vrf) error {
 			}
 		}
 	}
+	// Handover vrf routes into route manager to manage it.
+	for _, route := range vrf.routes {
+		vrfm.RouteManager.Add(route)
```

Comment:
It uses RouteManager's Add/Del APIs which are asynchronous, shouldn't consider for synchronous APIs as similar to VRF manager ?